### PR TITLE
Add Exercise 5

### PR DESCRIPTION
**Pull request by matthewhannay567**

### CODE REVIEW

File: assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Exercise 5.1
+This exercise covers how to create a launch file for node.
+
+# Exercise Objective
+Create a `.launch` file which starts the `printing_node`, which is already created for you.
+
+A launch file structured with XML (like HTML): it is composed of tags, which have attributes and
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md:7]: Can we link to a resource with the basic syntax, then I would also explicitly explain how to comment out some lines

File: assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+There is also an optional attribute `output`, which determines where the node should
+print messages. For this exercise, you want `output="screen"`
+
+The complete tag should look something like this:
+
+```xml
+<node pkg="package_name" type="node_type" name="descriptive name" output="screen">
+
+</node>
+```
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md]: Can we include the entire launch file for clarity?

File: assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+```xml
+<launch>
+
+</launch>
+```
+
+To make a launch file start node, you need a `<node>` tag inside the `<launch>` tags.
+The node tag requires three attributes:
+* `pkg`: which package the node can be found in
+* `type`: the type of node to launch, this is the string value put in `ros::init`
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md]: This should be better explained. It is the executable name for the node that is configured in cmake, I think you can make them look at a cmake file to make them understand this connection. Have them change the node.

File: assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* `name`: the name of this individual node. There can be multiple nodes with the same type,
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md]: Mention how this has semantic meaning but nothing that is set in stone by the build process. It is completely arbitrary.

File: assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+but no two nodes can have the same name.
+
+There is also an optional attribute `output`, which determines where the node should
+print messages. For this exercise, you want `output="screen"`
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_1/exercise_5_1.md]: I would have them run the node without output="screen" and with to get the idea of what is printed with each and what is not.

File: assignments/week_5/week_5_exercises/exercise_5_2/exercise_5_2.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# Exercise 5.2
+This exercise covers how to use launch parameters in launch files and in nodes.
+
+# Exercise Objective
+Right now `printing_node` has some issues with it. It has two arbitrary values:
+* the string being printed
+* the number of times that string is printed
+
+These values would be better suited as launch parameters, and not hardcoded.
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_2/exercise_5_2.md:9]: TODO, if not covered in video this needs to be explained more.

File: assignments/week_5/week_5_exercises/exercise_5_2/exercise_5_2.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+* `name`: the name of the parameter.
+* `value`: the value of the parameter. It's type is automatically determined depending on the input.
+
+Add two `param` tags to `launch_node.launch` in between the `<node>` tags, one that determines
+the message being printed, and another which determines how many times it is printed.
+
+# Parameters in a node
+To actually use the parameters defined in the launch file, we need to use the `NodeHandle` 
+and it's member function `getParam`. This function takes in two arguments:
+* the name of the parameter. This is actually the "path" name of the parameter, but because
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/exercise_5_2/exercise_5_2.md:28]: TODO is this explained in the video?

File: assignments/week_5/week_5_exercises/package.xml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  <!-- Use buildtool_depend for build tool packages: -->
+  <!--   <buildtool_depend>catkin</buildtool_depend> -->
+  <!-- Use exec_depend for packages you need at runtime: -->
+  <!--   <exec_depend>message_runtime</exec_depend> -->
+  <!-- Use test_depend for packages you need only for testing: -->
+  <!--   <test_depend>gtest</test_depend> -->
+  <!-- Use doc_depend for packages you need only for building documentation: -->
+  <!--   <doc_depend>doxygen</doc_depend> -->
+  <buildtool_depend>catkin</buildtool_depend>
+
```
Comment (JasonGibson274) [inline assignments/week_5/week_5_exercises/package.xml:53]: roscpp?

**Review by JasonGibson274 — COMMENTED** [review]
I think you should add a section where you have to use a couple common things that we tend to do in launch file. Specifically my suggestions are 

1. I would include a part where they need to have a specific comment in the launch file.

2. I think at least mentioning how to debug parameter namespace issues with rosparam list and get/set would  be a useful thing to cover.

**Review by kylestach — APPROVED** [review]

**Pull request merged**